code/game/objects/items/stacks/tape.dm

```diff
@@ -19,29 +19,43 @@
 	var/list/conferred_embed = EMBED_HARMLESS
 	var/overwrite_existing = FALSE
 
-/obj/item/stack/sticky_tape/afterattack(obj/item/I, mob/living/user)
+/obj/item/stack/sticky_tape/afterattack(obj/item/I, mob/living/user, proximity)
+	if(!proximity)
+		return
+
 	if(!istype(I))
 		return
 
+	if(I.embedding)
+		to_chat(user, "<span class='warning'>[I] is already capable of sticking to people!</span>")
+		return
+
 	if(I.embedding && I.embedding == conferred_embed)
 		to_chat(user, "<span class='warning'>[I] is already coated in [src]!</span>")
 		return
 
 	user.visible_message("<span class='notice'>[user] begins wrapping [I] with [src].</span>", "<span class='notice'>You begin wrapping [I] with [src].</span>")
 
 	if(do_after(user, 30, target=I))
-		use(1)
+		if(!proximity)
```

Comment:
proximity isn't a live value that could have changed, and do_after already fails if either the user or the target (the thing being taped) are moved or interrupted

Reply:
Nyoops